src/products/products-dynamodb.handler.ts

```diff
@@ -0,0 +1,99 @@
+import * as AWS from 'aws-sdk';
+
+const dynamodb = new AWS.DynamoDB.DocumentClient();
+
+export const getProductsList = async (_event: any = {}): Promise<any> => {
+  try {
+    const productsResult = await dynamodb.scan({ TableName: process.env.PRODUCTS_TABLE_NAME || '' }).promise();
+    const stockResult = await dynamodb.scan({ TableName: process.env.STOCK_TABLE_NAME || '' }).promise();
+
+    const products = productsResult.Items || [];
+    const stock = stockResult.Items || [];
+
+    const productList = products.map((product) => {
+      const stockItem = stock.find((item: any) => item.product_id === product.id);
```

Comment:
be aware, in case of production code and big data tables, this would lead to (O)n2 time complexity